Extension.php

```diff
@@ -11,10 +12,40 @@ public function initialize()
     {
         $this->addCss('assets/extension.css');
         $this->addJavascript('assets/start.js', true);
+
+        /*
+         * Routes:
+         * Here you can register new routes in your Bolt application.
+         * The first route will be handles in this Extension class,
+         * then we switch to an extra Controller class for the routes.
+         * */
+
+        // register route for all GET requests on '/example/url' that will be handled in this class ($this) in the 'routeExampleUrl' function
+        $this->app->get("/example/url", array($this, 'routeExampleUrl'))->bind('example-url');
+
+        // new instance of our example Controller
+        $exampleController = new ExampleController($this->app, $this->config);
+
+        // register route for all GET requests on '/example/url/in/controller' that will be handled in the ExampleController class in the 'exampleUrl' function
+        $this->app->get("/example/url/in/controller", array($exampleController, 'exampleUrl'))->bind('example-url-controller');
+
+        // {we register a new route stuff from below} ... and return a JSON string as response.
+        $this->app->get("/example/url/json", array($exampleController, 'exampleUrlJson'))->bind('example-url-json');
+
+        // {we register a new route stuff from below} ... with an url parameter whose value will be returned as a JSON response.
+        $this->app->get("/example/url/parameter/{id}", array($exampleController, 'exampleUrlWithParameter'))->bind('example-url-parameter');
     }
```

Comment:
@GawainLync vs. @PhillippOhlandt timing race condition detected :grin: 

> And where can I say which method it should use for the route

That is what connect if for, those calls to `->get()` `->post()` and `->match()` assume that the routes are based on what you specify in the `->mount()` call